### PR TITLE
feat/add razorpay

### PR DESCRIPTION
**Pull request by Suryansh777777**
Resolve Issue  #306
@VineeTagarwaL-code 

https://github.com/user-attachments/assets/1fbb504d-c289-45b7-b0aa-b2badfab60f1

Special Thanks to @itsflash10⚡

### CODE REVIEW

**Comment by VineeTagarwaL-code** [comment]
Great work buddy, reviewing the code right now check your discord dm lets hop on a call too

**Comment by VineeTagarwaL-code** [comment]
fix the build and conflicts please

**Comment by VineeTagarwaL-code** [comment]
Build errors still exist, please solve them ... need to merge this asap

**Comment by Suryansh777777** [comment]
You'll have to add env variables for Razorpay in github actions
and in the workflow yaml

**Comment by aakash2330** [comment]
@Suryansh777777 build is failing

File: src/components/job-form.tsx [hunk 1]
```diff
@@ -111,31 +114,16 @@ const PostJobForm = () => {
...
-    } catch (_error) {
-      toast({
-        title: 'Something went wrong will creating job',
-        description: 'Internal server error',
-        variant: 'destructive',
-      });
-    }
+    data.companyLogo = (await submitImage(file)) ?? '';
+    await processPayment({
+      amount: 1000,
```
Comment (aakash2330) [inline src/components/job-form.tsx:119]: keep it in env

File: src/hooks/useRazorPay.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    const razorpay = await intializeRazorpay();
+    if (!razorpay) {
+      toast({
+        title: 'Failed to initialize Razorpay',
+        variant: 'destructive',
+      });
+      return;
+    }
+
+    const order = await fetch('/api/order', {
```
Comment (aakash2330) [inline src/hooks/useRazorPay.ts:38]: can we turn these in actions ?

File: src/hooks/useRazorPay.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      handler: () => {
+        toast({
+          title: 'Payment Successful Thank you!',
+          variant: 'success',
+        });
+        successCallback && successCallback();
+      },
+    };
+
+    const paymentObject = new (window as any).Razorpay(options);
```
Comment (aakash2330) [inline src/hooks/useRazorPay.ts:85]: use correct types

File: src/hooks/useRazorPay.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import { toast } from '@/components/ui/use-toast';
+import { JobPostSchemaType } from '@/lib/validators/jobs.validator';
+
+const intializeRazorpay = () => {
+  return new Promise((resolve, reject) => {
+    const script = document.createElement('script');
```
Comment (aakash2330) [inline src/hooks/useRazorPay.ts:6]: use a next-script tag , it'll be cleaner (refer @ui//gmaps-autosuggest

File: src/components/job-form.tsx [hunk 5]
```diff
@@ -470,18 +458,21 @@ const PostJobForm = () => {
...
-          </div>
         </form>
       </Form>
 
       <div className="bg-gray-900 p-6 rounded-lg w-[37rem] mx-auto text-gray-300">
         <h2 className="text-lg font-semibold mb-4 text-gray-300">Payment</h2>
-        <Button className="w-full rounded-full mt-4">
-          Continue to Payment
+
+        <Button
```
Comment (aakash2330) [inline src/components/job-form.tsx:467]: can we put this button inside the form as a submit

File: src/app/api/verify/route.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  );
+
+  if (!isPaymentValid) {
+    return NextResponse.json(
+      { error: 'Payment not verified. Payment signature invalid' },
+      { status: 404 }
+    );
+  }
+
+  const jobData = jsonBody.payload.payment.entity.notes;
```
Comment (aakash2330) [inline src/app/api/verify/route.ts:27]: we have lodash library in deps , lets leverage the _.get function to access nested values

File: src/app/api/verify/route.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+  try {
+    const response = await createJob(jobData);
+
+    if (!response.status) {
+      return NextResponse.json({ error: response.message }, { status: 404 });
+    }
+
+    const transactionData = {
+      razorpayOrderId: jsonBody.payload.payment.entity.order_id,
```
Comment (aakash2330) [inline src/app/api/verify/route.ts:37]: same as above comment

File: src/app/api/order/route.ts [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    return NextResponse.json(
+      {
+        id: order.id,
+        currency: order.currency,
+        amount: order.amount,
+      },
+      { status: 200 }
+    );
+  } catch (error) {
+    return NextResponse.json({ error }, { status: 500 });
```
Comment (aakash2330) [inline src/app/api/order/route.ts:37]: we're serializing each api reponse to a certain shape , refer jobs/actions.ts

**Comment by chaitanyashetty47** [comment]
hey is this issue still up? can i work on it?

**Comment by VineeTagarwaL-code** [comment]
not for now we dont need it anymore

**Pull request closed without merging**